Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,256 @@
+/*global define*/
+define([
+        './Check',
+        './defined',
+        './defineProperties',
+        './Heap',
+        './isBlobUri',
+        './isDataUri',
+        './RequestState',
+        '../ThirdParty/when'
+    ], function(
+        Check,
+        defined,
+        defineProperties,
+        Heap,
+        isBlobUri,
+        isDataUri,
+        RequestState,
+        when) {
+    'use strict';
+
+    /**
+     * Tracks the number of active requests and prioritizes incoming requests.
+     *
+     * @exports RequestScheduler
+     *
+     * @private
+     */
+    function RequestScheduler() {
+    }
+
+    /**
+     * The maximum number of simultaneous active requests. Un-throttled requests do not observe this limit.
+     * @type {Number}
+     * @default 50
+     */
+    RequestScheduler.maximumRequests = 50;
```

Comment:
For now this option doesn't actually have much effect since terrain/imagery providers are throttled to 6 per server. This is more of a safety net if we decide to eventually remove the `throttleByServer` option for http2 providers so as not to overload the browser.

For now this class and these options should just stay private.